crates/ide/src/syntax_highlighting/highlight.rs

```diff
@@ -137,7 +137,7 @@ fn punctuation(
         }
         (T![!], MACRO_RULES) => HlPunct::MacroBang.into(),
         (T![!], NEVER_TYPE) => HlTag::BuiltinType.into(),
-        (T![!], PREFIX_EXPR) => HlOperator::Logical.into(),
+        (T![!], PREFIX_EXPR) => HlOperator::Negation.into(),
```

Comment:
```suggestion
        (T![!], PREFIX_EXPR) => HlOperator::Negation.into() | HlOperator::Logical.into(),
```
We should include both.

Reply:
I also hesitated to know whether `Negation` was actually disjoint from `Logical` (since they are now two separate variants of the `HlOperator` enum).

I think we could find reasons for both. My reason for "it is OK to dissociate them" is the fact that `!` is unary while `&&`and `||` are binary, so they may be seen as quite a different class of operators.

Reply:
Nothing in the name "logical" implies a binary operator. And in fact, the boolean negation operator is a logical operator by [common definition](https://en.wikipedia.org/wiki/Logical_connective).

However you're right that there is currently no way to combine operator tags, and implementing that won't be trivial (although it's possible), therefore I'll accept this PR.